h2/src/main/org/h2/command/dml/Merge.java

```diff
@@ -323,6 +323,10 @@ public void prepare() {
                 column.getSQL(builder, HasSQL.DEFAULT_SQL_FLAGS).append("=?");
             }
         }
+        if (!hasColumn) {
+            throw DbException.getSyntaxError(sqlStatement, sqlStatement.length(),
+                    "Valid MERGE INTO statement with at least one updatable column");
```

Comment:
with -> without

Reply:
It is what we expect and not what we found.

Reply:
Yes, we expect to "have at least one updateable column", but we did not find that.

So we should say
   "Valid MERGE INTO statement requires at least one updatable column"
or
    "MERGE INTO statement without at least one updatable column"

Reply:
The whole error message is:
`Syntax error in SQL statement "MERGE INTO T0(C0) KEY(C1) VALUES (1)[*]"; expected "Valid MERGE INTO statement with at least one updatable column"`

Reply:
Ah, you are correct

Reply:
It don't want to allocate a new error code for the normal error message for this specific corner case in non-standard command. This command has significant design problems, it can't handle `GENERATED ALWAYS` **identity** columns and **generated** columns well.